lib/c-api/Cargo.toml

```diff
@@ -53,7 +53,7 @@ wat = ["wasmer/wat"]
 wasi = ["wasmer-wasi", "typetag", "serde"]
 engine = []
 middlewares = ["wasmer-middlewares"]
-deprecated = ["libffi"]
+deprecated = ["libffi", "wasmer/experimental-reference-types-extern-ref"]
```

Comment:
Ummm I'm not sure if this was really intended?
We should have a new feature for reference types: `reference-types = ["wasmer/reference-types"]`

Reply:
We can make a separate feature for enabling it but it's not currently usable in the C API so it won't do anything.

This feature was needed to get the deprecated C API to compile, though after the latest edits I made it actually might not be necessary anymore.